### DISCUSSION TITLE
Accelerometer data not updating?

### DISCUSSION THREAD

**Discussion by globedasher** [efernau/rot8#41] [Q&A]
Hello,

I tried rot8 on Manjaro Sway and got no update to my display. I checked out the code and noted the `/sys/bus/iio/devices/iio:device*/in_accel_*_raw` path to what seems to be the data location. I ran a `watch -d cat /sys/bus/iio/devices/iio:device0/*` and saw no change to the data in all files there on device orientation changes. I inserted some print statements in rot8 and confirmed it was seeing the same lack of data changes from the `/sys/bus` files. 

I installed `iio-sensor-proxy` and ran `monitor-sensor` and received updates on orientation changes, and I also see data changes if I run `cat /dev/iio:device0 | xxd`. I read about what files need to be present in `/sys/bus/iio/devices/iio:device0/` to enable the buffer and all are present based on what I read. Now I'm lost. Any ideas on what I should check next or something I overlooked?

Thanks in advance.

**Comment by efernau** [comment]
Hey,

which device do you use?
If you have data in `/sys/bus/iio/devices/iio:device*/in_accel_*_raw` rot8 should work. 
Depending on what data is output there `--normalization-factor` can help you. 
You may need to play with the normalization factor (try multiples of 10) to get the accelerometer readings to calculate right.
Did you also specify your display with `--display`?

**Comment by globedasher** [comment]
Oops, I meant to include that. I am using an Acer Spin 5. It is a 2-in-1 laptop.
The data in `/sys/bus/iio/devices/iio:device*/in_accel_*_raw` is not changing. I believe that is my main issue.

Adding `--normalization-factor` did not produce different results.

I have tried `--display eDP-1` which is what my display is listed as with `swaymsg -t get_outputs` but no change. I  added a `println!("{}", output.name);` in the code at line 76. The output from that is `eDP-1` when I allow rot8 to auto-detect the display.

I added print statements of the variables `x_clean, y_clean, z_clean` near line 310 as those seem to be raw data, and those are always zero values when running.

If I translate the code rotation command to `swaymsg output eDP-1 transform <degrees>`, I can rotate the screen and touch input.

Thank you again for your time! :)

**Comment by efernau** [comment]
Do I understand you correctly, that `/sys/bus/iio/devices/iio:device*/in_accel_*_raw` has no output or does not change the output?
What is the output of `iio-sensor-proxy`?
Which package did you install, `community/iio-sensor-proxy` or `aur/iio-sensor-proxy-git`, try the other?
Which kernel do you use? `uname -r`
Do you have [Linux kernel sysfs IIO](https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/Documentation/ABI/testing/sysfs-bus-iio) module in your kernel?  
Are you sure your accelerometer is not broken, did it work with other OS?

**Reply by globedasher** [reply]
Correct that `/sys/bus/iio/devices/iio:device*/in_accel_*_raw` has no change. I get a lot of values if I do a `watch -d cat /sys/bus/iio/devices/iio:device*/in_accel_*_raw` but I never see it change.
I cannot run `iio-sensor-proxy` directly but I can run the `monitor-sensor` program installed with that package and it shows:

![swappy-20230123_155402](https://user-images.githubusercontent.com/18297955/214178592-97df5a99-525d-41a5-9881-31cbc9eb4a41.png)

I am using the community version of iio-sensor-proxy (installed with `pacman -S iio-sensor-proxy`)

Kernel is 5.15.85-1-MANJARO

`Do you have [Linux kernel sysfs IIO](https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/Documentation/ABI/testing/sysfs-bus-iio) module in your kernel?` The dirs seem to be there, but I'm not savvy enough to say for sure. 

I switched to the Manjaro Sway community build from Manjaro i3 community build. I was able to get screen rotation working in i3 on this same machine.
